ID: 16153
Title: Internat error when reading excel file
Description:
### What happens?

Internat error `Attempted to access index 4490 within vector of size 4490` when reading excel file

### To Reproduce

```
select 
    * 
from read_xlsx('NUTS2021-NUTS2024.xlsx', sheet='NUTS-3 Typologies')  
```

the file is a public file that can be downloaded from [https://ec.europa.eu/eurostat/documents/345175/629341/NUTS2021-NUTS2024.xlsx](https://ec.europa.eu/eurostat/documents/345175/629341/NUTS2021-NUTS2024.xlsx)

Other sheets of the file seem to work, but I don't see anything specific that would cause the crash. It has cyrillic characters but other working sheets too. 


it works if I use st_read instead of read_excel

### OS:

Apple M1, Macos Sequoia

### DuckDB Version:

1.2.0

### DuckDB Client:

sql

### Hardware:

_No response_

### Full Name:

Luc Martinon

### Affiliation:

freelancer

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [ ] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [ ] Yes, I have